src/clients/HubPoolClient.ts

```diff
@@ -807,9 +816,7 @@ export class HubPoolClient extends BaseAbstractClient {
     }
     const update = await this._update(eventsToQuery);
     if (!update.success) {
-      // This failure only occurs if the RPC searchConfig is miscomputed, and has only been seen in the hardhat test
-      // environment. Normal failures will throw instead. This is therefore an unfortunate workaround until we can
-      // understand why we see this in test. @todo: Resolve.
+      this.isUpdated = update.reason === UpdateFailureReason.AlreadyUpdated;
```

Comment:
Does this mean that we will go from being updated to not updated in some cases? I think previously, we were crashing on RPC errors instead of returning success=false.

IIUC, that means that RPC errors will show up as the hub pool not being updated, causing the program to crash when that is checked. This will also mean the error will say that the hub pool was not updated.

Is that desired?

The concern is two-fold:
1. Less transparency about _what_ the RPC error is when it crashes.
2. Error message about hub pool not being updated is a bit of a red herring.

Thoughts? Do you agree with this analysis or am I misunderstanding the change?

Reply:
Fair call RE suppressing RPC errors. I think we'd be logging these at the Provider level, but it's not a good idea to rely on that, and if an update fails on ConfigStore or HubPool then we probably want to bail anyway. I've updated the PR to back that part out, such that RPC provider exceptions won't be caught anymore.